FTBFS on aarch64 due to type mismatch: c_char may be i8 or u8

**Describe the bug**

A scratch build on Fedora failed on aarch64 https://kojipkgs.fedoraproject.org//work/tasks/2034/117682034/build.log

**To Reproduce**

Run `cargo test -p chewing_capi` on aarch64

```
error[E0308]: mismatched types
  --> capi/src/io.rs:92:81
   |
92 |         .and_then(|data| str::from_utf8(unsafe { mem::transmute::<&[i8], &[u8]>(data) }).ok())
   |                                                  ------------------------------ ^^^^ expected `&[i8]`, found `&[u8]`
   |                                                  |
   |                                                  arguments to this function are incorrect
   |
   = note: expected reference `&[i8]`
              found reference `&[u8]`
note: function defined here
  --> /builddir/build/BUILD/rustc-1.78.0-src/library/core/src/intrinsics.rs:1419:12
For more information about this error, try `rustc --explain E0308`.
```

**Expected behavior**

Test should pass

**Platform (please complete the following information):**
 - OS: Fedora
 - Version: 0.8.0

**Additional context**

https://github.com/rust-lang/rust/issues/79089

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,7 +16,7 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [ubuntu-latest, windows-latest, macos-latest, macos-14]
         preset:
           - rust-release
           - rust-with-sqlite-release
@@ -30,7 +30,7 @@ jobs:
 
       - if: ${{ matrix.os == 'ubuntu-latest' }}
         run: sudo apt-get install ninja-build
-      - if: ${{ matrix.os == 'macos-latest' }}
+      - if: ${{ matrix.os == 'macos-latest' || matrix.os == 'macos-14' }}
         run: brew install ninja
       - if: ${{ matrix.os == 'windows-latest' }}
         run: choco install ninja
@@ -51,23 +51,6 @@ jobs:
       - name: Test
         run: |
           ctest --test-dir build --output-on-failure
-
-  aarch64-smoke:
-    runs-on: ubuntu-latest
-    name: Aarch64 cross-compile smoke test
-
-    steps:
-      - uses: actions/checkout@v3
-      - name: Setup rustup
-        run: |
-          rustup default ${{ env.RUST_VERSION }}
-          rustup target add aarch64-unknown-linux-gnu
-          rustup update
-
-      - name: Build
-        run: |
-          cargo build --target=aarch64-unknown-linux-gnu
-          cargo build -p chewing_capi --target=aarch64-unknown-linux-gnu
 
   coverage:
     strategy:
@@ -126,7 +109,7 @@ jobs:
     if: ${{ always() }}
     runs-on: ubuntu-latest
     name: Final Results
-    needs: [build, aarch64-smoke]
+    needs: [build]
     steps:
       - run: exit 1
         # see https://stackoverflow.com/a/67532120/4907315

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -52,6 +52,23 @@ jobs:
         run: |
           ctest --test-dir build --output-on-failure
 
+  aarch64-smoke:
+    runs-on: ubuntu-latest
+    name: Aarch64 cross-compile smoke test
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup rustup
+        run: |
+          rustup default ${{ env.RUST_VERSION }}
+          rustup target add aarch64-unknown-linux-gnu
+          rustup update
+
+      - name: Build
+        run: |
+          cargo build --target=aarch64-unknown-linux-gnu
+          cargo build -p chewing_capi --target=aarch64-unknown-linux-gnu
+
   coverage:
     strategy:
       matrix:
@@ -109,7 +126,7 @@ jobs:
     if: ${{ always() }}
     runs-on: ubuntu-latest
     name: Final Results
-    needs: [build]
+    needs: [build, aarch64-smoke]
     steps:
       - run: exit 1
         # see https://stackoverflow.com/a/67532120/4907315

--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,9 @@
+What's New in libchewing 0.8.1 (May 15, 2024)
+---------------------------------------------------------
+
+* Buf fixed
+  - Fail to build on aarch64 due to type mismatch. (introduced in v0.8.0)
+
 What's New in libchewing 0.8.0 (May 14, 2024)
 ---------------------------------------------------------
 

--- a/capi/src/io.rs
+++ b/capi/src/io.rs
@@ -89,7 +89,7 @@ unsafe fn slice_from_ptr_with_nul<'a>(ptr: *const c_char) -> Option<&'a [c_char]
 
 unsafe fn str_from_ptr_with_nul<'a>(ptr: *const c_char) -> Option<&'a str> {
     unsafe { slice_from_ptr_with_nul(ptr) }
-        .and_then(|data| str::from_utf8(unsafe { mem::transmute::<&[i8], &[u8]>(data) }).ok())
+        .and_then(|data| str::from_utf8(unsafe { mem::transmute::<&[c_char], &[u8]>(data) }).ok())
 }
 
 #[no_mangle]

--- a/tests/testhelper/src/path.rs
+++ b/tests/testhelper/src/path.rs
@@ -83,7 +83,7 @@ pub unsafe extern "C" fn find_path_by_files(
     -1
 }
 
-unsafe fn files_ptr_to_slice(files: *const *const i8) -> Vec<Cow<'static, str>> {
+unsafe fn files_ptr_to_slice(files: *const *const c_char) -> Vec<Cow<'static, str>> {
     let len = {
         let mut i = 0;
         while unsafe { !files.add(i).read().is_null() } {

